### PR TITLE
Graph export without graphviz

### LINKED ISSUE #1722: Allow exporting graph as displayed in GUI
**Is your feature request related to a problem? Please describe.**

Current graph export directly uses methods provided by r2. This means that  graphviz needs to be installed. This is non obvious and somewhat inconvenient on non Linux platforms.

**Describe the solution you'd like**

Add an option for exporting graph image using the same rendering and layout code used for displaying it in GUI. This has benefit of not requiring graphviz to be installed and resulting image matching GUI including highlighted blocks. This should be easily doable by passing appropriate QPaintingDevice to QPainter.

R2 commands can still be use for exporting to .dot and other formats which can't be easily generated in the way described above.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by karliss**
**Detailed description**

Some refactoring in graph widget so that drawing functions don't depend on current widget state and can be reused for drawing to image file.

Graph image export now has entries for exporting as png, jpg or svg using Cutter's drawing code.  R2+Graphviz based export is still there but new export is preferred for the supported formats unless Graphviz based export is explicitly selected.

 New file save dialog with multiple type support.  Unfortunately can't be done with native file save dialogs at least through Qt. I am a bit surprised that doesn't have proper builtin file save dialog that supports multiple types.  In QFileDialog type selection functions only as filter even in save mode. The new save dialog should prevent some of the confusion that could happen when exported file type differed from selected one.

**Test plan (required)**

* Test that all formats create image with correct format (don't trust extension)
* Test type detection from extension
* Try type detection with invalid extension

exportg

**Closing issues**
Closes #1722

### CODE REVIEW

File: src/widgets/DisassemblerGraphView.cpp [hunk 1]
```diff
@@ -1028,45 +1012,76 @@ void DisassemblerGraphView::blockTransitionedTo(GraphView::GraphBlock *to)
...
+
+    case GraphExportType::GVJson:
+    case GraphExportType::GVGif:
+    case GraphExportType::GVPng:
+    case GraphExportType::GVJpeg:
+    case GraphExportType::GVPostScript:
+    case GraphExportType::GVSvg:
+        TempConfig tempConfig;
+        tempConfig.set("graph.gv.format", selectedType.extension);
+        qWarning() << Core()->cmd(QString("agfw \"%1\" @ $FB").arg(filePath));
```
Comment (xarkes) [inline src/widgets/DisassemblerGraphView.cpp]: It's maybe not important because "hard to trigger", but isn't it vulnerable to command injection?

Comment (karliss) [reply src/widgets/DisassemblerGraphView.cpp]: That's the code that was there before my changes, but I would be happy to improve it. Do we have appropriate API for that? 

One mitigation approach I considered some time ago was having separate methods for executing single r2 command and evaluating arbitrary r2 shell script. In 90% of cases it sufficient to execute only single command.

**Comment by karliss** [comment]
If `$FB` works the way I think it does, there will probably be problems with exporting from unsynchronized graph.

File: src/widgets/DisassemblerGraphView.cpp (hunk 1 above)
Comment (ITAYC0HEN) [reply src/widgets/DisassemblerGraphView.cpp]: There is cmdRaw. But as for @thestr4ng3r there are things to impove still, like replaceing "\" with "\\"

https://github.com/radareorg/cutter/blob/9e680d772b34fb7eba829ef353ac04fb67662ed8/src/core/Cutter.cpp#L318-L323

Comment (karliss) [reply src/widgets/DisassemblerGraphView.cpp]: Ok I'll use that for now.  I assume you meant escaping backslash but github ate half of them.

File: src/widgets/DisassemblerGraphView.cpp [hunk 2]
```diff
@@ -1028,45 +1012,76 @@ void DisassemblerGraphView::blockTransitionedTo(GraphView::GraphBlock *to)
...
+        });
     }
 
-    QFileDialog dialog(this, tr("Export Graph"));
-    dialog.setAcceptMode(QFileDialog::AcceptSave);
-    dialog.setFileMode(QFileDialog::AnyFile);
-    dialog.setNameFilters(filters);
+    MultitypeFileSaveDialog dialog(this, tr("Export Graph"));
+    dialog.setTypes(types);
     dialog.selectFile("graph");
```
[the remark this answers is not in the document]
Comment (karliss) [reply src/widgets/DisassemblerGraphView.cpp]: Done.

**Pull request merged**